core/services/keystore/keys/export_test.go

```diff
@@ -0,0 +1,101 @@
+package keys_export_test
+
+import (
+	"testing"
+
+	"github.com/smartcontractkit/chainlink/core/services/keystore/keys/csakey"
+	"github.com/smartcontractkit/chainlink/core/services/keystore/keys/ethkey"
+	"github.com/smartcontractkit/chainlink/core/services/keystore/keys/ocrkey"
+	"github.com/smartcontractkit/chainlink/core/services/keystore/keys/p2pkey"
+	"github.com/smartcontractkit/chainlink/core/services/keystore/keys/solkey"
+	"github.com/smartcontractkit/chainlink/core/services/keystore/keys/terrakey"
+	"github.com/smartcontractkit/chainlink/core/services/keystore/keys/vrfkey"
+	"github.com/smartcontractkit/chainlink/core/utils"
+	"github.com/stretchr/testify/require"
+)
+
+type KeyType interface {
+	ethkey.KeyV2 | csakey.KeyV2 | p2pkey.KeyV2 | vrfkey.KeyV2 | ocrkey.KeyV2 | solkey.Key | terrakey.Key
+	ToEncryptedJSON(password string, scryptParams utils.ScryptParams) (export []byte, err error)
+	String() string
+}
+
+func TestKeyExportImport(t *testing.T) {
+	t.Parallel()
+
+	tests := []struct {
+		name     string
+		testCase func(*testing.T)
+	}{
+		{"ethkey", runTestCase[ethkey.KeyV2]},
+		{"csakey", runTestCase[csakey.KeyV2]},
+		{"p2pkey", runTestCase[p2pkey.KeyV2]},
+		{"vrfkey", runTestCase[vrfkey.KeyV2]},
+		{"ocrkey", runTestCase[ocrkey.KeyV2]},
+		{"solkey", runTestCase[solkey.Key]},
+		{"terrakey", runTestCase[solkey.Key]},
+	}
+
+	for _, test := range tests {
+		test := test
+
+		t.Run(test.name, func(t *testing.T) {
+			t.Parallel()
+			test.testCase(t)
+		})
+	}
+}
+
+func runTestCase[T KeyType](t *testing.T) {
+	key, err := createKey[T]()
```

Comment:
nit/ Would a different letter read a little easier for these, since they are nearby `testing.T`s?
```suggestion
func runTestCase[K KeyType](t *testing.T) {
	key, err := createKey[K]()
```

Reply:
Agreed, fixed.